src/jit/morph.cpp

```diff
@@ -8038,8 +8043,8 @@ void Compiler::fgMorphTailCall(GenTreeCall* call, void* pfnCopyArgs)
 #endif // _TARGET_X86_
 
 #if defined(_TARGET_X86_)
-        // When targeting x86, the runtime requires that we perforrm a null check on the `this` argument before tail
-        // calling to a virtual dispatch stub. This requirement is a consequence of limitations in the runtime's
+        // When targeting arm or x86, the runtime requires that we perform a null check on the `this` argument before
```

Comment:
I think that the runtime requires this always, not just for arm or x86.

I believe that amd64 is just getting lucky because of VSD stubs on amd64 always get null check (even when they are not tailcalls).

Reply:
Is it required on unix as well? We saw this failure only on Windows.

Reply:
I think we should have it on Unix as well.

Reply:
> We saw this failure only on Windows.

I could easily convince myself that #17933 is related to this.  The back trace I was debugging included a VSD stub

Reply:
Do you also see tailcall helper on the stack? This problem is only going to happen with tailcalls - they are super rare.